### PR TITLE
Remove passive task resend logic

### PR DESCRIPTION
**Pull request by xwduan**
## What changed?

Remove passive task resend logic
## Why?

Resend is best effort
## How did you test it?

unit test
## Potential risks

passive task may disappear
## Documentation

no document
## Is hotfix candidate?

no

### CODE REVIEW

File: service/history/ndc_standby_task_util.go [hunk 1]
```diff
@@ -97,74 +100,81 @@ func standbyTimerTaskPostActionTaskDiscarded(
...
+func isWorkflowExistOnSource(
+	ctx context.Context,
+	taskInfo tasks.Task,
+	logger log.Logger,
+	currentCluster string,
+	clientBean client.Bean,
+	registry namespace.Registry,
+) bool {
+	workflowKey := taskWorkflowKey(taskInfo)
+	remoteClusterName, err := getRemoteClusterName(
```
Comment (yux0) [inline service/history/ndc_standby_task_util.go:112]: nit: sourceCluster to match the method name?

File: service/history/ndc_standby_task_util.go [hunk 2]
```diff
@@ -97,74 +100,81 @@ func standbyTimerTaskPostActionTaskDiscarded(
...
+		workflowKey.GetNamespaceID(),
+	)
+	if err != nil {
+		return true
 	}
+	remoteAdminClient, err := clientBean.GetRemoteAdminClient(remoteClusterName)
+	if err != nil {
+		return true
+	}
+	_, err = remoteAdminClient.DescribeMutableState(ctx, &adminservice.DescribeMutableStateRequest{
```
Comment (yux0) [inline service/history/ndc_standby_task_util.go:124]: Could we use DescribeWorkflowExecution?

File: service/history/timer_queue_standby_task_executor.go [hunk 3]
```diff
@@ -166,7 +160,7 @@ func (t *timerQueueStandbyTaskExecutor) executeUserTimerTimeoutTask(
 				timerTask.GetVisibilityTime(),
 				timerSequenceID.Timestamp,
 			) {
-				return getHistoryResendInfo(mutableState)
+				return &struct{}{}, nil
```
Comment (yux0) [inline service/history/timer_queue_standby_task_executor.go:163]: nit: seem like we don't need extra data in the struct. could we just use a boolean?

Comment (xwduan) [reply service/history/timer_queue_standby_task_executor.go:163]: Current interface is taking an struct as input. I would love to keep the interface as is for now.

File: service/history/ndc_standby_task_util.go [hunk 4]
```diff
@@ -97,74 +100,81 @@ func standbyTimerTaskPostActionTaskDiscarded(
...
+			return false
+		}
+		logger.Error("Error describe mutable state from remote.",
+			tag.WorkflowNamespaceID(workflowKey.GetNamespaceID()),
+			tag.WorkflowID(workflowKey.GetWorkflowID()),
+			tag.WorkflowRunID(workflowKey.GetRunID()),
+			tag.ClusterName(remoteClusterName),
+			tag.Error(err))
+	}
+	return true
```
[the remark this answers is not in the document]
Comment (xwduan) [reply service/history/ndc_standby_task_util.go:142]: Yeah, this is the best effort check. If it is other error i.e internal error, then we should still execute the original logic(discard task with error and fire alarm).

**Review by hai719 — APPROVED** [review]

**Pull request merged**